sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -151,6 +151,7 @@ case class CatalogTable(
     tableType: CatalogTableType,
     storage: CatalogStorageFormat,
     schema: StructType,
+    originalSchema: Option[StructType] = None,
```

Comment:
The only difference between originalSchema and schema are the column names, right? If that is the case why not store the `source` column name in the schema field metadata?